packages/flutter/lib/src/material/carousel.dart

```diff
@@ -848,6 +896,13 @@ class _RenderSliverFixedExtentCarousel extends RenderSliverFixedExtentBoxAdaptor
     double itemExtent,
   ) {
     if (maxExtent > 0.0) {
+      if (infinite) {
+        // For infinite scrolling, calculate the max visible index
+        // based on the viewport size and allow negative indices
+        final double viewportItems = constraints.viewportMainAxisExtent / maxExtent;
+        return (scrollOffset / maxExtent + viewportItems).ceil();
+      }
```

Comment:
This seems not accurate as well. My understanding is the max index should just be scrollOffset / maxExtent - 1. `scrollOffset / maxExtent` is how many items we can include and `- 1` is to indicate the item index.

Reply:
Again, this was for bidirectional carousel. Have removed it.